cli/azd/pkg/tools/azcli/container_registry.go

```diff
@@ -93,46 +116,40 @@ func (crs *containerRegistryService) Login(ctx context.Context, subscriptionId s
 		// If that fails, attempt to get ACR credentials from the admin user
 		adminCreds, adminErr := crs.getAdminUserCredentials(ctx, subscriptionId, loginServer)
 		if adminErr != nil {
-			return fmt.Errorf("failed logging into container registry, token: %w, admin: %w", tokenErr, adminErr)
+			return nil, fmt.Errorf("failed logging into container registry, token: %w, admin: %w", tokenErr, adminErr)
 		}
 
 		dockerCreds = adminCreds
 	}
 
-	err := crs.docker.Login(ctx, dockerCreds.LoginServer, dockerCreds.Username, dockerCreds.Password)
-	if err != nil {
-		return fmt.Errorf(
-			"failed logging into docker registry %s: %w",
-			loginServer,
-			err)
-	}
-
-	return nil
+	return dockerCreds, nil
 }
 
+// getTokenCredentials
 func (crs *containerRegistryService) getTokenCredentials(
 	ctx context.Context,
 	subscriptionId string,
 	loginServer string,
-) (*dockerCredentials, error) {
+) (*DockerCredentials, error) {
 	acrToken, err := crs.getAcrToken(ctx, subscriptionId, loginServer)
 	if err != nil {
 		return nil, fmt.Errorf("failed getting ACR token: %w", err)
 	}
 
-	return &dockerCredentials{
+	return &DockerCredentials{
 		Username:    "00000000-0000-0000-0000-000000000000",
```

Comment:
This doesn't align with [sdk-container-build's description of SDK_CONTAINER_REGISTRY_UNAME](https://github.com/dotnet/sdk-container-builds/blob/main/docs/RegistryAuthentication.md#authenticating-to-container-registries). Curious if the documentation is amiss.

> This should be the username for the registry. If the password for the registry is a token, then the username should be "".

Reply:
The all zeros username comes from this documentation: https://learn.microsoft.com/en-us/azure/container-registry/container-registry-authentication?tabs=azure-cli#individual-login-with-microsoft-entra-id

I'm not sure if "token" in sdk-container-build documentation is referring to a subtly different type of token (since Docker itself seems to have some O-Auth based token login strategy, but it might be compatible out of the box with Entra ID)?

I'll add a comment here to link to the above documentation about the all zeros GUID.